schema/manager.go

```diff
@@ -393,39 +446,85 @@ func (manager *Manager) LoadExtensions(extensions []*Resource) error {
 
 //LoadNamespaces register namespaces by db object
 func (manager *Manager) LoadNamespaces(namespaces []*Resource) error {
+	manager.mu.Lock()
+	defer manager.mu.Unlock()
+
 	for _, namespaceData := range namespaces {
 		namespace, err := NewNamespace(namespaceData.Data())
 		if err != nil {
 			return err
 		}
-		manager.RegisterNamespace(namespace)
+		manager.registerNamespace(namespace)
 	}
 
 	return nil
 }
 
 //ClearExtensions clears extensions
 func (manager *Manager) ClearExtensions() {
-	manager.Extensions = []*Extension{}
+	manager.mu.Lock()
+	manager.Extensions = manager.Extensions[:0]
+	manager.mu.Unlock()
 }
 
+// UseGls if set to true makes GetManager return separate instances for
+// different goroutines trees, this is needed for running tests in parallel.
+var UseGls = false
```

Comment:
There are several problems with this:
1. Gls suggest using of concrete implementation (github.com/tylerb/gls), not the usage intent: one manager per goroutine
2. It's a global variable
3. It's a global variable used in multithreaded context and not protected for synchronization
4. Changing it at runtime yields unpredictable behaviours

I suggest adding a simple layer of abstraction (perhaps a Strategy) to configure global instance/one instances per goroutine before the first call to GetManager.